File: software/examples/ruby/example_scan_for_tags.rb
#!/usr/bin/env ruby
# -*- ruby encoding: utf-8 -*-

require 'tinkerforge/ip_connection'
require 'tinkerforge/bricklet_nfc_rfid'

include Tinkerforge

HOST = 'localhost'
PORT = 4223
UID = 'XYZ' # Change to your UID

ipcon = IPConnection.new # Create IP connection
nr = BrickletNFCRFID.new UID, ipcon # Create device object
current_tag_type = 0

ipcon.connect HOST, PORT # Connect to brickd
# Don't use device before ipcon is connected

# Register state changed callback
nr.register_callback(BrickletNFCRFID::CALLBACK_STATE_CHANGED) do |state, idle|
  if idle
    current_tag_type = (current_tag_type + 1) % 3
    nr.request_tag_id current_tag_type
  end

  if state == BrickletNFCRFID::STATE_REQUEST_TAG_ID_READY
    tag_type, tid_length, tid = nr.get_tag_id
    puts "Found tag of type #{tag_type} with ID " +
         "#{tid.take(tid_length).each.map { |b| b.to_s(16) }.join ' '}"
  end
end

nr.request_tag_id(BrickletNFCRFID::TAG_TYPE_MIFARE_CLASSIC)

puts 'Press key to exit'
$stdin.gets
ipcon.disconnect

File: software/examples/php/ExampleScanForTags.php
<?php

require_once('Tinkerforge/IPConnection.php');
require_once('Tinkerforge/BrickletNFCRFID.php');

use Tinkerforge\IPConnection;
use Tinkerforge\BrickletNFCRFID;

const HOST = 'localhost';
const PORT = 4223;
const UID = 'XYZ'; // Change XYZ to the UID of your NFC/RFID Bricklet

$tag_type = 0;

// Callback function for state changed callback
function cb_stateChanged($state, $idle, $user_data)
{
	$nr = $user_data;

	if($idle) {
		global $tag_type;
		$tag_type = ($tag_type + 1) % 3;
		$nr->requestTagID($tag_type);
	}

	if($state == BrickletNFCRFID::STATE_REQUEST_TAG_ID_READY) {
		$ret = $nr->getTagID();
		echo "Found tag of type " . $ret["tag_type"] . " with ID [" . dechex($ret["tid"][0]);

		for($i = 1; $i < $ret["tid_length"]; $i++) {
			echo " " . dechex($ret["tid"][$i]);
		}

		echo "]\n";
	}
}

$ipcon = new IPConnection(); // Create IP connection
$nr = new BrickletNFCRFID(UID, $ipcon); // Create device object

$ipcon->connect(HOST, PORT); // Connect to brickd
// Don't use device before ipcon is connected

// Register state changed callback to function cb_stateChanged
$nr->registerCallback(BrickletNFCRFID::CALLBACK_STATE_CHANGED, 'cb_stateChanged', $nr);

// Start scan loop
$nr->requestTagID(BrickletNFCRFID::TAG_TYPE_MIFARE_CLASSIC);

echo "Press ctrl+c to exit\n";
$ipcon->dispatchCallbacks(-1); // Dispatch callbacks forever

?>
